### PR TITLE
[Backport] [2.x] Fixed misunderstanding message 'No OpenSearchException found' when detailed_error disabled

### PR DESCRIPTION
**Pull request by xuezhou25**
Signed-off-by: Xue Zhou <xuezhou@amazon.com>

### Description
Backport #4669 

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Fixed the SnapshotsInProgress error during index deletion ([#4570](https://github.com/opensearch-project/OpenSearch/pull/4570))
 - [Bug]: Fixed invalid location of JDK dependency for arm64 architecture([#4613](https://github.com/opensearch-project/OpenSearch/pull/4613))
 - [Bug]: Alias filter lost after rollover ([#4499](https://github.com/opensearch-project/OpenSearch/pull/4499))
+- Fixed misunderstanding message "No OpenSearchException found" when detailed_error disabled ([#4669](https://github.com/opensearch-project/OpenSearch/pull/4669))
```
Comment (dreamer-89) [inline CHANGELOG.md:57]: 1. Need to change PR number 4669 -> 4708
2. We can update the wording to make it more clear.

**Review by dreamer-89 — CHANGES_REQUESTED** [review]

**Comment by dreamer-89** [comment]
The PR is failing because of recent version bump in 1.x from 1.3.6 -> 1.3.7 as part of https://github.com/opensearch-project/OpenSearch/pull/4701. We need to update the same in 2.x. Once that PR is merged to 2.x, you need to rebase against 

```
1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':distribution:bwc:maintenance:buildBwcLinuxTar'.
> Building 1.3.6 didn't generate expected file /var/jenkins/workspace/gradle-check/search/distribution/bwc/maintenance/build/bwc/checkout-1.3/distribution/archives/linux-tar/build/distributions/opensearch-min-1.3.6-SNAPSHOT-linux-x64.tar.gz

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
```

**Comment by dreamer-89** [comment]
@xuezhou25 : Can you please rebase against latest 2.x branch ? It should fix the bwc test failures.

File: server/src/main/java/org/opensearch/OpenSearchException.java [hunk 2]
```diff
@@ -594,16 +594,14 @@ public static void generateFailureXContent(XContentBuilder builder, Params param
...
             Throwable t = e;
             for (int counter = 0; counter < 10 && t != null; counter++) {
                 if (t instanceof OpenSearchException) {
-                    message = t.getClass().getSimpleName() + "[" + t.getMessage() + "]";
                     break;
                 }
                 t = t.getCause();
             }
-            builder.field(ERROR, message);
+            builder.field(ERROR, ExceptionsHelper.summaryMessage(t));
```
Comment (nknize) [inline server/src/main/java/org/opensearch/OpenSearchException.java:604]:
```suggestion
            builder.field(ERROR, ExceptionsHelper.summaryMessage(e));
```
`t` is null, which results in a more confusing `Internal error` exception even when a useful `IllegalArgumentException` is thrown. `e` should be passed to `summaryMessage` so a more useful message is thrown. This already resulted in a [reproducible test failure](https://build.ci.opensearch.org/job/gradle-check/4166/) which was fixed in #4702.

**Review by nknize — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/opensearch/OpenSearchException.java (hunk 2 above)
Comment (reta) [reply server/src/main/java/org/opensearch/OpenSearchException.java:604]: @nknize I think the logic behind this code is to find the `OpenSearchException` down the cause chain (which is `t`). May be in this case we should do:

```
builder.field(ERROR, ExceptionsHelper.summaryMessage(t != null ? t : e));
```

?

Comment (xuezhou25) [reply server/src/main/java/org/opensearch/OpenSearchException.java:604]: Thanks for reviewing. I will do this fix

**Comment by andrross** [comment]
@xuezhou25 Can you rebase and fix changelog conflicts?

File: server/src/main/java/org/opensearch/OpenSearchException.java (hunk 2 above)
Comment (tlfeng) [reply server/src/main/java/org/opensearch/OpenSearchException.java]: I agree with the suggested change from @reta . The original logic is to find the `OpenSearchException` down the cause chain, and show its error message.

Comment (andrross) [reply server/src/main/java/org/opensearch/OpenSearchException.java]: The `ExceptionsHelper.summaryMessage(e)` version is what [exists on main currently](https://github.com/opensearch-project/OpenSearch/blob/f0511311e3166cda7ef47fc5ba7e45b61240b94b/server/src/main/java/org/opensearch/OpenSearchException.java#L591). We should probably keep them consistent and then follow up with a change on main to do the `ExceptionsHelper.summaryMessage(t != null ? t : e)` version (and then backport that PR as well).

Comment (tlfeng) [reply server/src/main/java/org/opensearch/OpenSearchException.java]: @andrross Make sense, thank you! 😄

**Review by tlfeng — APPROVED** [review]

**Pull request merged**